Rules/UseDeclaredVarsMoreThanAssignments.cs

```diff
@@ -204,6 +198,24 @@ private IEnumerable<DiagnosticRecord> AnalyzeScriptBlockAst(ScriptBlockAst scrip
                 }
             }
 
+            // Detect usages of Get-Variable
+            var getVariableCmdletNamesAndAliases = Helper.Instance.CmdletNameAndAliases("Get-Variable");
+            IEnumerable<Ast> getVariableCommandAsts = scriptBlockAst.FindAll(testAst => testAst is CommandAst commandAst &&
+                getVariableCmdletNamesAndAliases.Contains(commandAst.GetCommandName(), StringComparer.OrdinalIgnoreCase), true);
+            foreach (CommandAst getVariableCommandAst in getVariableCommandAsts)
+            {
+                var commandElements = getVariableCommandAst.CommandElements.ToList();
+                // The following extracts the variable name only in the simplest possibe case 'Get-Variable variableName'
+                if (commandElements.Count == 2 && commandElements[1] is StringConstantExpressionAst constantExpressionAst)
+                {
+                    var variableName = constantExpressionAst.Value;
+                    if (assignments.ContainsKey(variableName))

```

Comment:
a comment indicating this is case insensitive? (I missed that until I saw the entire file - maybe it's ok)

Reply:
Comments get out of date, therefore a rename of the dictionary variable `assignments` to `assignmentsDictionary_OrdinalIgnoreCase` would maybe be better? WDYT?

Reply:
Comments can get out of date quickly, therefore I renamed the dictionary variable `assignments` to `assignmentsDictionary_OrdinalIgnoreCase`.